accounts/service.go

```diff
@@ -93,12 +96,15 @@ func (s *InterceptorService) Start(lightningClient lndclient.LightningClient,
 	s.routerClient = routerClient
 	s.checkers = NewAccountChecker(s, params)
 
+	s.isEnabled = true
```

Comment:
I agree that this would be a much cleaner solution. Unfortunately though, we have calls in the `Start` function that requires that `isRunning` returns true to not error, which is why we need to set it as enabled in the start of the function.

1. The `Start` function calls `TrackPayment` which requires that the service is running to add the payment for tracking to the `lndclient.RouterClient`. Potentially this could be changed, and only require the service to be running when the actual payment update is returned by the `lndclient.RouterClient`. Though that feels a bit wrong, so I'd prefer to keep it as is unless you disagree.

2. The `Start` function spawns a goroutine that calls `invoiceUpdate` that definitely requires that the service is running when being executed. This could potentially be solved though by letting the `Start` function acquire the store lock throughout the execution of the function.

Let me know if you think I should change this, or if the current version of the `Start` function is ok :).

Reply:
gotcha! thanks for the explanation :)